src/node_buffer.cc

```diff
@@ -40,18 +40,19 @@
 
 #define THROW_AND_RETURN_IF_OOB(r)                                          \
   do {                                                                      \
-    if (!(r))                                                               \
+    if ((r).IsNothing()) return;                                            \
```

Comment:
Nit: If my understanding is correct, this `return;` would make the `Copy` return with `undefined`. Would it be better to include a test which checks that?

Reply:
It would not return any value – this is the case where coercing to integer throws an exception (i.e. the last part of the test)